### PR TITLE
Add leaderboards

### PR DESCRIPTION
**Pull request by SaraVieira**
- Checks if you already set a name, if not asks you a name and saves it in localstorage
- Adds you to the leaderboard
- Shows the number you are

Issues:
- Restart does not work idk why

### CODE REVIEW

**Comment by Gusted** [comment]
This conflicting with #99

**Comment by SaraVieira** [comment]
yeah, I don't know what to do about that, I talked to paul about making this yesterday and noticed that came in when I was almost done

**Comment by Gusted** [comment]
Maybe time to open up issues to track features and make sure not 2 people are working on the same thing.

**Comment by njm222** [comment]
My bad I didn't know you were working on it too. I can look into the issues in a bit, pretty sure I resolved the first one already.

**Comment by SaraVieira** [comment]
Im super sorry too, did not want to step on anyone's toes :(

The env thing is fixed

**Comment by njm222** [comment]
No worries :) I'm just glad it's almost done

**Comment by SaraVieira** [comment]
Definiutly do think we can use your PR and add auth in the future

**Comment by njm222** [comment]
Yeah I can merge it in afterwards! (Looking into the restart issue now)

**Review by drcmda — APPROVED** [review]
very happy with this. 🎉 @njm222 you found the restart problem? i think it can go in then. and like you said, we can merge your auth part afterwards. thank you @SaraVieira !

File: src/ui/Finished.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { useStore, mutation } from '../store'
+import { createClient } from '@supabase/supabase-js'
+import { useState } from 'react'
+
+export const supabase = createClient(import.meta.env['VITE_SUPABASE_URL'], import.meta.env['VITE_SUPABASE_ANON_KEY'])
```
Comment (njm222) [inline src/ui/Finished.jsx:5]: fine for now.. I'll extract this out when I merge auth

**Review by njm222 — APPROVED** [review]

**Pull request merged**